Source/DafnyLanguageServer/Language/DiagnosticErrorReporter.cs

```diff
@@ -82,25 +85,66 @@ public class DiagnosticErrorReporter : ErrorReporter {
       );
     }
 
-    private static IEnumerable<DiagnosticRelatedInformation> CreateDiagnosticRelatedInformationFor(IToken token, string message) {
+    public static readonly string PostConditionFailingMessage = new EnsuresDescription().FailureDescription;
+    private readonly string entryDocumentsource;
+
+    public static string FormatRelated(string related) {
+      return $"Could not prove: {related}";
+    }
+
+    private IEnumerable<DiagnosticRelatedInformation> CreateDiagnosticRelatedInformationFor(IToken token, string message) {
+      var (tokenForMessage, inner) = token is NestedToken nestedToken ? (nestedToken.Outer, nestedToken.Inner) : (token, null);
+      if (tokenForMessage is RangeToken range) {
+        var rangeLength = range.EndToken.pos + range.EndToken.val.Length - range.StartToken.pos;
+        if (message == PostConditionFailingMessage) {
```

Comment:
I'm not thrilled with this way of detecting the type of message, but I suspect it may be necessary for now, and that we'd have to do a lot of refactoring to be able to do it in a nicer way.